meshtastic/interfaces/ble/interface.py

```
@@ -0,0 +1,1899 @@
+                            logger.error("Fatal BLE read error after retries: %s", e)
+                            if not self._state_manager._is_closing:
+                                self.close()
+                            return
+                    except (RuntimeError, OSError) as e:
+                        # Treat these as fatal errors that should close the interface
+                        logger.error("Fatal error in BLE receive thread: %s", e)
+                        if not self._state_manager._is_closing:
+                            self.close()
+                        return
+                    except (
+                        Exception
```

Ruff (0.15.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)